pkg/distributor/dataobj_tee.go

```diff
@@ -50,12 +52,18 @@ func (c *DataObjTeeConfig) Validate() error {
 // DataObjTee is a tee that duplicates streams to the data object topic.
 // It is a temporary solution while we work on segmentation keys.
 type DataObjTee struct {
-	cfg          *DataObjTeeConfig
-	limitsClient *ingestLimits
-	limits       Limits
-	kafkaClient  *kgo.Client
-	resolver     *SegmentationPartitionResolver
-	logger       log.Logger
+	cfg              *DataObjTeeConfig
+	rateStore        *segmentationKeyRateStore
+	distributorsRing *ring.Ring
+	limits           Limits
+	kafkaClient      KafkaProducer
+	resolver         *SegmentationPartitionResolver
+	logger           log.Logger
+
+	// A simple cache to avoid repeatedly calling GetHealthy on the ring.
```

Comment:
Is GetHealthy an expensive operation?

Reply:
Yeah fairly https://github.com/grafana/dskit/blob/0f495c43f84b/ring/ring.go#L644